### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by imhilla**
Kindly review codes for the enumerable methods project of the ruby section.
This project entailed coming up with replica methods of the ruby inbuilt method. I learned about the Enumerable module that gets mixed into the Array and Hash classes (among others) and provides you with lots of handy iterator methods.
-I created a script file to house the methods and used the IRB to test them.
-Added new methods onto the existing Enumerable module.
-This project is one of those projects that introduces one to backend and would be happy to get good feedback from the TSE
Kindly note this is a second pull request, I had issues with git-flow. [Here's](https://github.com/imhilla/enumerable-methods/pull/1) the previous one

Regards.
Hillary Kiptoo

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
```
Comment (Forison) [inline enumerables.rb:4]: * [ ] There should be a working `my_each`  identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

### replace ```my_each``` with ```each``` and compare results in the example below.

```p [1,2,3,4,5].my_each{ |a| a}```

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      when 'Hash' then yield(keys[i], self[keys[i]])
+      when 'Array' then yield(self[i])
+      when 'Range' then yield(to_a[i])
+      end
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
```
Comment (Forison) [inline enumerables.rb]: * [ ] There should be a working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

### replace ```my_each_with_index``` with ```each_with_index``` and compare results in the example below.
``` p [1,2,3,4,5].my_each_with_index{ |a,b| b} ```

**Review by Forison — CHANGES_REQUESTED** [review]
## Status => Require changes❌
# 🙋‍♂️ Hillary,
You have done a good job so far. Here are a few changes we have to make to get this project approved😎 .
> This looks like a re-review from my dashboard. Please do not open a pull request for a re-review unless you did not follow the required Gitflow.
* [ ] Kindly add how to set up and use the project on a local machine in the readme file.
* [ ] Kindly place your test examples in a separate file.
* [ ] Check below comments for other changes you have to equally address
## That is it kindly make these changes and submit the project for re-review🤗.
Happy coding
Addo

**Review by nikasulo — CHANGES_REQUESTED** [review]
# Great job! @imhilla 
Please check on this in a few minutes for some feedback!

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Review by nikasulo — CHANGES_REQUESTED** [review]
# Great work @imhilla 
Please consider my comments below, make the necessary changes and request a re-review! 

#### my_map
Please consider the screenshot below. `my_map` should execute only the proc when a proc and block are given! Please correct this and make another review request! I wish you the very best and see you at the top :rocket: 
screenshot

#### Stickler errors
You have done very nicely by including stickler in your project as requested! However, you need to ensure that all errors it throws are appropriately addressed before you request a PR review. Please look into this, address appropriately, and request another review. 

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello @imhilla ,  You have done a great job with this project. :clap:
#### You can go ahead and merge !!!

**Pull request merged**